lib/primitives/txmeta.js

```diff
@@ -161,8 +161,12 @@ class TXMeta {
     json.time = this.time;
     json.confirmations = 0;
 
-    if (chainHeight != null)
-      json.confirmations = chainHeight - this.height + 1;
+    if (chainHeight != null) {
+      if (this.height == -1)
+        json.confirmations = 0;
+      else
+        json.confirmations = chainHeight - this.height + 1;
+      }
```

Comment:
The indentation is off and makes it read as there is a`}` missing. If the height is `-1`, it will always have `confirmations` of `0`, regardless if `chainHeight` is `null`, so it could be:

```js
if (this.height == -1) {
  json.confirmations = 0;
} else if (chainHeight != null) {
  json.confirmations = chainHeight - this.height + 1;
}
```

Reply:
Actually, `json.confirmations` is already set to `0`, just above. So I think this can be:

```js
if (chainHeight != null && this.height != -1)
  json.confirmations = chainHeight - this.height + 1;
```